### ISSUE TITLE
Okada model -  horizontal deformation 

### ISSUE THREAD

**Issue by EranFru** [clawpack/geoclaw#360]
Hello everyone,

I use Geoclaw for modeling of tsunami waves propagation, for some time. Hope you could help me with the next issue.

I modeled a tsunami generated by earthqauke. For my understanding the Okada model used in Geoclaw take in consider only vertical deformation, because the horizontal component can be neglected generally. 
Since the source parameters of the modeled EQ. is strike slip- the horizontal contribution to the vertical displacement has to be added (Tanioka and Satake, 1996).

My question is : Do the Okada model of Geocalw take in consider the horizontal contribution to the vertical displacement? If yes, where can i find the relevant documentation to prove my strike-slip EQ. model is legitimate?

Thanks!

Eran.

**Comment by rjleveque** [comment]
The Okada model in the current release does not include horizontal displacements.  We have a version that also computes horizontal displacements from some work a couple years ago but I guess that never got cleaned up and merged in.  One version can be found [here](https://github.com/cjvogl/finite_volume_seismic_model/blob/v1.0.0/3d/dtopotools_horiz_okada_and_1d.py) from [this paper](http://staff.washington.edu/rjl/pubs/VoglLeVeque2017/index.html) with @cjvogl, and a slightly different version in the [seismic repository](https://github.com/rjleveque/seismic/blob/master/src/python/seismic/dtopotools_horiz_okada_and_1d.py).  I'm not sure offhand which is better.

In the meantime the `dtopotools.py` in GeoClaw has evolved further and we may have also made some further improvements to the horizontal motion part.   I'll look into this more, but in the meantime perhaps one of those codes is useful.  @dsrim also recently added a triangular subfault version of Okada to the version in 5.5.0, but I don't think that includes horizontal motion.

The horizontal displacements for this half-space code would also have to be combined with the actual bathymetry in the region of interest in order to translate the horizontal motion into the corresponding vertical motion, which is how it would probably be used.  Of course horizontal motion of a steep continental slope might also impart horizontal momentum to the fluid, but that would be harder to incorporate into a shallow water equation model such as GeoClaw, since momentum is only imparted near the bottom, not throughout the water column.

**Comment by dsrim** [comment]
@rjleveque Actually... I did include the surface horizontal motion (dtopo.dX, dtopo.dY) in triangular version - however as you mentioned I did not anticipate its immediate use since we will need to combine it with bathymetry. I also did not thoroughly test the horizontal components, so it must be used with some caution.

**Comment by EranFru** [comment]
Thank you very much for the quick responses.

**Comment by mandli** [comment]
@rjleveque has volunteered to create a PR for this.

**Comment by fabian-kutschera** [comment]
Hi everybody,
I am curious if there has been an update regarding the Tanioka filter and accounting for the contribution of the horizontal displacement (based on the bathymetry) to the vertical displacement.
Has the functionality changed since this issue was opened?

Best
Fabian

**Comment by rjleveque** [comment]
@fabian-kutschera: No, sorry!  We've been too busy with some other issues and enhancements, but thanks for bringing this up again since it's good to bump it up higher in the to-do list.
